File: app/src/main/java/com/hfad/myferma/incubator/editDayIncubatorFragment.java
package com.hfad.myferma.incubator;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import com.google.android.material.appbar.MaterialToolbar;
import com.google.android.material.textfield.TextInputLayout;
import com.hfad.myferma.R;
import com.hfad.myferma.db.MyFermaDatabaseHelper;
import com.hfad.myferma.db.MydbManagerMetod;

public class editDayIncubatorFragment extends Fragment implements View.OnClickListener {

    private TextInputLayout tempInput, dampInput, overInput, airingInput;
    private Button update_button;
    private TextView textUnit;
    private MydbManagerMetod mydbManager;
    private String[] massId, massTempDamp, massOver, massAiring;
    private int dataFragment;
    private String idFragment, tempFragment, dampFragment, overFragment, airingFragment;
    private  MyFermaDatabaseHelper myDB;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View layout = inflater.inflate(R.layout.fragment_edit_day_incubator, container, false);
        myDB = new MyFermaDatabaseHelper(getActivity());
        MaterialToolbar appBar = getActivity().findViewById(R.id.topAppBar);
        appBar.getMenu().findItem(R.id.delete).setVisible(false);

        Bundle bundle = this.getArguments();
        if (bundle != null) {
            dataFragment = bundle.getInt("data");
            massId = bundle.getStringArray("id");
            massTempDamp = bundle.getStringArray("tempDamp");
            massOver = bundle.getStringArray("over");
            massAiring = bundle.getStringArray("airing");
        }

        textUnit = layout.findViewById(R.id.text_unit);
        tempInput = layout.findViewById(R.id.discAdd_input);
        dampInput = layout.findViewById(R.id.dayAdd_input);
        overInput = layout.findViewById(R.id.mountAdd_input);
        airingInput = layout.findViewById(R.id.yearAdd_input);

        textUnit.setText("День " + String.valueOf(dataFragment));
        tempInput.getEditText().setText(massTempDamp[dataFragment]);
        dampInput.getEditText().setText(massTempDamp[dataFragment+30]);
        overInput.getEditText().setText(massOver[dataFragment]);
        airingInput.getEditText().setText(massAiring[dataFragment]);

        update_button = layout.findViewById(R.id.update_button);
        update_button.setOnClickListener(this);

        return layout;
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.update_button:
                upDate(v);
                break;
        }
    }

    public void upDate(View view) {

        massTempDamp[dataFragment] = tempInput.getEditText().getText().toString().trim().replaceAll(",", ".").replaceAll("[^\\d.]", "");
        massTempDamp[dataFragment+30] = dampInput.getEditText().getText().toString().trim().replaceAll("\\D+", "");
        massOver[dataFragment] = overInput.getEditText().getText().toString().trim();
        massAiring[dataFragment] = airingInput.getEditText().getText().toString().trim();

        //убираем ошибку
        tempInput.setErrorEnabled(false);
        dampInput.setErrorEnabled(false);
        overInput.setErrorEnabled(false);
        airingInput.setErrorEnabled(false);

        //вывод ошибки
        if (tempInput.getEditText().getText().toString().equals("") || dampInput.getEditText().getText().toString().equals("") ||
                overInput.getEditText().getText().toString().equals("") || airingInput.getEditText().getText().toString().equals("")) {
            if (tempInput.getEditText().getText().toString().equals("")) {
                tempInput.setError("Укажите температуру!");
                tempInput.getError();
            }
            if (dampInput.getEditText().getText().toString().equals("")) {
                dampInput.setError("Укажите влажность");
                dampInput.getError();
            }
            if (overInput.getEditText().getText().toString().equals("")) {
                overInput.setError("Укажите кол-во переворотов");
                overInput.getError();
            }
            if (airingInput.getEditText().getText().toString().equals("")) {
                airingInput.setError("Укажите кол-вл проветриваний");
                airingInput.getError();
            }
        } else {
            myDB.updateIncubator(massTempDamp, massId[0]);
            myDB.updateIncubatorOver(massOver, massId[0]);
            myDB.updateIncubatorAiring(massAiring, massId[0]);
        }
    }
}

File: app/src/main/java/com/hfad/myferma/incubator/editDayIncubatorFragment.kt
package com.hfad.myferma.incubator

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.google.android.material.appbar.MaterialToolbar
import com.google.android.material.textfield.TextInputLayout
import com.hfad.myferma.R
import com.hfad.myferma.db.MyFermaDatabaseHelper

class editDayIncubatorFragment : Fragment(), View.OnClickListener {
    private lateinit var tempInput: TextInputLayout
    private lateinit var dampInput: TextInputLayout
    private lateinit var overInput: TextInputLayout
    private lateinit var airingInput: TextInputLayout
    private lateinit var updateButton: Button
    private lateinit var textUnit: TextView

    private var massId = mutableListOf<String>()
    private var massTemp = mutableListOf<String>()
    private var massDamp = mutableListOf<String>()
    private var massOver = mutableListOf<String>()
    private var massAiring = mutableListOf<String>()
    private var dataFragment: Int = 0
    private var dayColum = 0
    private var allListFragment : Boolean = false

    private lateinit var myDB: MyFermaDatabaseHelper

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        val layout: View = inflater.inflate(R.layout.fragment_edit_day_incubator, container, false)
        myDB = MyFermaDatabaseHelper(requireContext())

        val appBar: MaterialToolbar = requireActivity().findViewById(R.id.topAppBar)
        appBar.menu.findItem(R.id.delete).isVisible = false
        appBar.menu.findItem(R.id.filler).isVisible = false
        appBar.setNavigationOnClickListener {
            requireActivity().supportFragmentManager.popBackStack()
        }

        var massId2 = arrayOf<String>()
        var massTemp2 = arrayOf<String>()
        var massDamp2 = arrayOf<String>()
        var massOver2 = arrayOf<String>()
        var massAiring2 = arrayOf<String>()


        val bundle: Bundle? = arguments
        if (bundle != null) {
            dataFragment = bundle.getInt("data")
            massId2 = bundle.getStringArray("id")!!
            massTemp2 = bundle.getStringArray("temp")!!
            massDamp2 = bundle.getStringArray("damp")!!
            massOver2 = bundle.getStringArray("over")!!
            massAiring2 = bundle.getStringArray("airing")!!
            allListFragment = bundle.getBoolean("allList")

        }
        dayColum = dataFragment-1

        massId = massId2.toMutableList()
        massTemp = massTemp2.toMutableList()
        massDamp = massDamp2.toMutableList()
        massOver = massOver2.toMutableList()
        massAiring = massAiring2.toMutableList()

        textUnit = layout.findViewById(R.id.text_unit)
        tempInput = layout.findViewById(R.id.discAdd_input)
        dampInput = layout.findViewById(R.id.dayAdd_input)
        overInput = layout.findViewById(R.id.mountAdd_input)
        airingInput = layout.findViewById(R.id.yearAdd_input)

        textUnit.text = "День $dataFragment "
        tempInput.editText!!.setText(massTemp[dayColum])
        dampInput.editText!!.setText(massDamp[dayColum])
        overInput.editText!!.setText(massOver[dayColum])
        airingInput.editText!!.setText(massAiring[dayColum])

        updateButton = layout.findViewById(R.id.update_button)
        updateButton.setOnClickListener(this)
        return layout
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.update_button -> upDate(v)
        }
    }

    private fun upDate(view: View?) {
        massTemp[dayColum] =
            tempInput.editText!!.text.toString().trim { it <= ' ' }
                .replace(",".toRegex(), ".").replace("[^\\d.]".toRegex(), "")

        massDamp[dayColum] =
            dampInput.editText!!.text.toString().trim { it <= ' ' }
                .replace("\\D+".toRegex(), "")

        massOver[dayColum] =
            overInput.editText!!.text.toString().trim { it <= ' ' }
        massAiring[dayColum] =
            airingInput.editText!!.text.toString().trim { it <= ' ' }

        //убираем ошибку
        tempInput.isErrorEnabled = false
        dampInput.isErrorEnabled = false
        overInput.isErrorEnabled = false
        airingInput.isErrorEnabled = false

        //вывод ошибки
        if (((tempInput.editText!!
                .text.toString() == "") || (dampInput.editText!!
                .text.toString() == "") || (overInput.editText!!
                .text.toString() == "") || (airingInput.editText!!.text
                .toString() == ""))
        ) {
            if ((tempInput.editText!!.text.toString() == "")) {
                tempInput.error = "Укажите температуру!"
                tempInput.error
            }
            if ((dampInput.editText!!.text.toString() == "")) {
                dampInput.error = "Укажите влажность"
                dampInput.error
            }
            if ((overInput.editText!!.text.toString() == "")) {
                overInput.error = "Укажите кол-во переворотов"
                overInput.error
            }
            if ((airingInput.editText!!.text.toString() == "")) {
                airingInput.error = "Укажите кол-вл проветриваний"
                airingInput.error
            }
        } else {
            myDB.updateIncubatorTemp(massTemp, massId[0])
            myDB.updateIncubatorDamp(massDamp, massId[0])
            myDB.updateIncubatorOver(massOver, massId[0])
            myDB.updateIncubatorAiring(massAiring, massId[0])

            replaceFragment()



        }
    }

    private fun replaceFragment()
    {
        val bundle: Bundle = Bundle()
        var fragment: Fragment = AllListFragment()

        if (allListFragment) {
            bundle.putInt("data",dataFragment)
            bundle.putStringArray("id", massId.toTypedArray())
            bundle.putStringArray("temp", massTemp.toTypedArray())
            bundle.putStringArray("damp", massDamp.toTypedArray())
            bundle.putStringArray("over", massOver.toTypedArray())
            bundle.putStringArray("airing", massAiring.toTypedArray())
        } else {
            bundle.putString("name", massId[1])
            bundle.putString("type", massId[2])
            bundle.putString("data", massId[3])
            bundle.putString("id", massId[0])
            fragment = NowIncubatorFragment()
        }

        requireActivity().supportFragmentManager.beginTransaction()
            .replace(R.id.conteiner, fragment, "visible_fragment")
            .addToBackStack(null)
            .commit()
        fragment.arguments = bundle
    }

}